components/eamxx/src/physics/mam/eamxx_mam_constituent_fluxes_interface.cpp

```diff
@@ -0,0 +1,291 @@
+    add_field<Updated>(cld_nmr_field_name, scalar3d_mid, n_unit, grid_name);
+
+    for(int a = 0; a < mam_coupling::num_aero_species(); ++a) {
+      // (interstitial) aerosol tracers of interest: mass (q) mixing ratios
+      const char *int_mmr_field_name =
+          mam_coupling::int_aero_mmr_field_name(mode, a);
+      if(strlen(int_mmr_field_name) > 0) {
+        add_field<Updated>(int_mmr_field_name, scalar3d_mid, q_unit, grid_name,
+                           "tracers");
+      }
+      // (cloudborne) aerosol tracers of interest: mass (q) mixing ratios
+      // NOTE: DO NOT add cld borne aerosols to the "tracer" group as these are
+      // NOT advected
+      const char *cld_mmr_field_name =
+          mam_coupling::cld_aero_mmr_field_name(mode, a);
+      if(strlen(cld_mmr_field_name) > 0) {
+        add_field<Updated>(cld_mmr_field_name, scalar3d_mid, q_unit, grid_name);
+      }
+    }  // end for loop num species
+  }    // end for loop for num modes
+
+  for(int g = 0; g < mam_coupling::num_aero_gases(); ++g) {
+    const char *gas_mmr_field_name = mam_coupling::gas_mmr_field_name(g);
+    add_field<Updated>(gas_mmr_field_name, scalar3d_mid, q_unit, grid_name,
+                       "tracers");
+  }  // end for loop num gases
+
+}  // set_grid
+
+// ================================================================
+//  REQUEST_BUFFER_SIZE_IN_BYTES
+// ================================================================
+// ON HOST, returns the number of bytes of device memory needed by the above
+// Buffer type given the number of columns and vertical levels
+size_t MAMConstituentFluxes::requested_buffer_size_in_bytes() const {
+  return mam_coupling::buffer_size(ncol_, nlev_);
+}
+
+// ================================================================
+//  INIT_BUFFERS
+// ================================================================
+// ON HOST, initializes the Buffer type with sufficient memory to store
+// intermediate (dry) quantities on the given number of columns with the given
+// number of vertical levels. Returns the number of bytes allocated.
+void MAMConstituentFluxes::init_buffers(
+    const ATMBufferManager &buffer_manager) {
+  EKAT_REQUIRE_MSG(
+      buffer_manager.allocated_bytes() >= requested_buffer_size_in_bytes(),
+      "Error! Insufficient buffer size.\n");
+
+  size_t used_mem =
+      mam_coupling::init_buffer(buffer_manager, ncol_, nlev_, buffer_);
+  EKAT_REQUIRE_MSG(
+      used_mem == requested_buffer_size_in_bytes(),
+      "Error! Used memory != requested memory for MAMConstituentFluxes.");
+}
+
+// ================================================================
+//  INITIALIZE_IMPL
+// ================================================================
+void MAMConstituentFluxes::initialize_impl(const RunType run_type) {
+  // ---------------------------------------------------------------
+  // Input fields read in from IC file, namelist or other processes
```

Comment:
A question related to process coupling - if multiple processes have this initialization step, which one of them will read the tracer concentration from the initial data? Is it the first one called?

Reply:
Yes. Each process can be run individually as well, so we have to keep this code but only the first process reads it. Once it is loaded in memory, other processes can use it.